ecs-cli/modules/clients/aws/secretsmanager/client.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package secretsmanager
+
+import (
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+)
+
+// SMClient defines methods for interacting with the SecretsManagerAPI interface
+type SMClient interface {
+	CreateSecret(secretsmanager.CreateSecretInput) (*secretsmanager.CreateSecretOutput, error)
+	ListSecrets(*string) (*secretsmanager.ListSecretsOutput, error)
+}
+
+type secretsManagerClient struct {
+	client secretsmanageriface.SecretsManagerAPI
```

Comment:
I'm a little confused by this field -- is there a reason we can't have the `secretsmanager.SecretsManager` as the type for client rather than the whole sdk interface for the secrets manager? In the former, you could leverage the [factory method](https://github.com/aws/aws-sdk-go/blob/master/service/secretsmanager/service.go#L47) defined in the SDK; alternatively you could have this client be the `SMClient` interface you define. Then the sdk client would satisfy that interface implicity (similar to what's happening in #565).

Reply:
I'm not sure I understand. Using the SDK interface [is](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/ecs/client.go#L68) [how](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/ec2/client.go#L34) [we](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/ssm/client.go#L44) [typically](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/cloudwatchlogs/client.go#L31) define the client type, since this is the interface [we use to generate](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/ecs/generate_mock.go#L17) the mock clients. 

I'm using this factory method to actually set the client on line 36. If we use `secretsmanager.SecretsManager` as the type on line 30 how would we generate and use a mock client in tests (since gomock needs an interface as the generation source)?

Reply:
Sorry, I got confused with something else. This LGTM :)